src/panels/Menu/TopMenuBar/TaskBarMenuChoices.tsx

```diff
@@ -0,0 +1,66 @@
+import { Dispatch, SetStateAction } from 'react';
+import { CheckboxIndicator, Menu } from '@mantine/core';
+
+import { ChevronRightIcon, TaskBarIcon } from '@/icons/icons';
+import { IconSize } from '@/types/enums';
+import { menuItemsData } from '@/windowmanagement/data/MenuItems';
+
+import { MenuWrapper } from './MenuWrapper';
+
+interface TaskBarChoicesProps {
+  visibleMenuItems: string[];
+  setVisibleMenuItems: Dispatch<SetStateAction<string[]>>;
+}
+
+export function TaskBarMenuChoices({
+  visibleMenuItems,
+  setVisibleMenuItems
+}: TaskBarChoicesProps) {
+  // Mapping menu item ids to boolean flag to determine their checked status
+  const checkedMenuItems: { [key: string]: boolean } = {};
+  visibleMenuItems.forEach((menuItemID) => {
+    checkedMenuItems[menuItemID] = true;
+  });
+
```

Comment:
For some reason I still find the way this is written very confusing... i think the perfomance loss is negligible if we just have a `includes` in later in the `checked={...}` prop. i guess it would be `checked={visibleMenuItems.includes(item.componentID)}`